sdk/examples/wallet/offline_signing/0_generate_address.rs

```diff
@@ -50,17 +51,18 @@ async fn main() -> Result<()> {
         .with_bip_path(Bip44::new(SHIMMER_COIN_TYPE))
         .finish()
         .await?;
-
     println!("Generated a new wallet");
 
-    write_wallet_address_to_file(&wallet).await
+    write_wallet_address_to_file(&wallet.address().await).await?;
+
+    Ok(())
 }
 
-async fn write_wallet_address_to_file(wallet: &Wallet) -> Result<()> {
+// Backups the Bech32 wallet address to JSON file.
```

Comment:
why the change?

Reply:
personal preference tbh ... the method name announces it will only require the wallet's address, so we're giving it just that, and not a reference to the wallet itself.

Reply:
Want me to change it back? I really don't mind if you think giving it a `&Wallet` is better.

Reply:
I don't really care, just wanted to be sure this was intentional 🤷‍♀️